### PR TITLE
Add `compose top` command

### PR DESCRIPTION
**Pull request by aiordache**
Implement `compose top`.

```

$ docker compose top
compose-cli_web_1
UID    PID     PPID    C    STIME   TTY   TIME       CMD
root   73864   73803   0    10:24   ?     00:00:00   nginx: master process nginx -g daemon off;   
101    74422   73864   0    10:24   ?     00:00:00   nginx: worker process                        

compose-cli_web_2
UID    PID     PPID    C    STIME   TTY   TIME       CMD
root   73863   73784   0    10:24   ?     00:00:00   nginx: master process nginx -g daemon off;   
101    74230   73863   0    10:24   ?     00:00:00   nginx: worker process                        

compose-cli_web_3
UID    PID     PPID    C    STIME   TTY   TIME       CMD
root   73862   73823   0    10:24   ?     00:00:00   nginx: master process nginx -g daemon off;   
101    74400   73862   0    10:24   ?     00:00:00   nginx: worker process                        

```
```
$ docker-compose top
compose-cli_web_1
UID     PID    PPID    C   STIME   TTY     TIME                        CMD                    
----------------------------------------------------------------------------------------------
root   73864   73803   0   10:24   ?     00:00:00   nginx: master process nginx -g daemon off;
101    74422   73864   0   10:24   ?     00:00:00   nginx: worker process                     

compose-cli_web_2
UID     PID    PPID    C   STIME   TTY     TIME                        CMD                    
----------------------------------------------------------------------------------------------
root   73863   73784   0   10:24   ?     00:00:00   nginx: master process nginx -g daemon off;
101    74230   73863   0   10:24   ?     00:00:00   nginx: worker process                     

compose-cli_web_3
UID     PID    PPID    C   STIME   TTY     TIME                        CMD                    
----------------------------------------------------------------------------------------------
root   73862   73823   0   10:24   ?     00:00:00   nginx: master process nginx -g daemon off;
101    74400   73862   0   10:24   ?     00:00:00   nginx: worker process 
```

### CODE REVIEW

File: local/compose/top.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+func (s *composeService) Top(ctx context.Context, projectName string, services []string) ([]compose.ContainerProcSummary, error) {
+	containers, err := s.apiClient.ContainerList(ctx, moby.ContainerListOptions{
+		Filters: filters.NewArgs(projectFilter(projectName)),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	ignore := func(string) bool {
```
Comment (ndeloof) [inline local/compose/top.go:36]: you don't need this, use `containers.filter(isService(services...))`

**Review by ndeloof — CHANGES_REQUESTED** [review]
use of `ignore` func can be avoided

File: cli/cmd/compose/top.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+				_, _ = fmt.Fprintf(w, strings.Repeat("%s\t", len(info))+"\n", info...)
+
+			}
+			fmt.Fprintln(w)
+		},
+			container.Titles...)
+		if err != nil {
+			return err
+		}
+	}
```
Comment (gtardif) [inline cli/cmd/compose/top.go:86]: You could extract this in a function to unit test it (will need a variable for stdout)

**Review by gtardif — CHANGES_REQUESTED** [review]
LGTM, but with a quick e2e test (after one of the existing compose up) for this command

**Review by gtardif — APPROVED** [review]

**Review by ndeloof — APPROVED** [review]

**Pull request merged**